src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -525,11 +525,25 @@ internal string ConstructName(TypeNameFormatFlags formatFlags)
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern void* _GetUtf8Name(RuntimeType type);
+        private static extern unsafe bool GetUtf8NameInternal(MethodTable* pMT, void** name);
 
-        internal static MdUtf8String GetUtf8Name(RuntimeType type)
+        internal static unsafe MdUtf8String GetUtf8Name(RuntimeType type)
         {
-            return new MdUtf8String(_GetUtf8Name(type));
+            TypeHandle th = type.GetNativeTypeHandle();
+            if (th.IsTypeDesc || th.AsMethodTable()->IsArray)
+            {
+                throw new ArgumentException(SR.Arg_InvalidHandle);
+            }
+
+            void* name;
+            if (!GetUtf8NameInternal(th.AsMethodTable(), &name))
+            {
+                throw new BadImageFormatException();
+            }
```

Comment:
```suggestion
            void* name = GetUtf8NameInternal(th.AsMethodTable());
            if (name is null)
            {
                throw new BadImageFormatException();
            }
```
Make the error handling uniform between types/methods/fields